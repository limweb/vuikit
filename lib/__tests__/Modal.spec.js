import { createVue, destroyVM, triggerKeyEvent, queryByTag, toJSON } from './_util'
import waitForUpdate from './_wait-for-update'

describe('Modal', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', () => {
    vm = createVue(`
      <vk-modal :large="true" :lightbox="true" :blank="true" :show="true">Modal</vk-modal>
    `)
    expect(toJSON(vm)).toMatchSnapshot()
  })

  it('adds classes to document on display', done => {
    vm = createVue(`
      <vk-modal :show="true" :transition="false"></vk-modal>
    `)
    waitForUpdate(() => {
      expect(document.documentElement.classList.contains('uk-modal-page')).toBeTruthy()
    }).then(done)
  })

  it('removes classes from document when hidden', done => {
    vm = createVue({
      template: `
        <vk-modal :show="show" :transition="false"></vk-modal>
      `,
      data: () => ({
        show: true
      })
    })
    vm.show = false
    waitForUpdate(() => {
      expect(document.documentElement.classList.contains('uk-modal-page')).toBeFalsy()
    }).then(done)
  })

  it('triggers event escKey when pressed ESC', done => {
    vm = createVue(`
      <vk-modal :show="true" :transition="false"></vk-modal>
    `)
    const cb = jest.fn()
    const modal = queryByTag(vm, 'vk-modal')
    modal.$on('keyEsc', cb)
    triggerKeyEvent(document.documentElement, 27)
    waitForUpdate(() => {
      expect(cb).toHaveBeenCalled()
    }).then(done)
  })

  it('triggers event clickOut when clicked outside the Modal dialog', done => {
    vm = createVue(`
      <vk-modal :show="true" :transition="false"></vk-modal>
    `)
    const cb = jest.fn()
    const modal = queryByTag(vm, 'vk-modal')
    modal.$on('clickOut', cb)
    modal.$el.click()
    waitForUpdate(() => {
      expect(cb).toHaveBeenCalled()
    }).then(done)
  })

  it('triggers event clickIn when clicked inside the Modal dialog', done => {
    vm = createVue(`
      <vk-modal :show="true" :transition="false"></vk-modal>
    `)
    const cb = jest.fn()
    const modal = queryByTag(vm, 'vk-modal')
    modal.$on('clickIn', cb)
    modal.$refs.dialog.click()
    waitForUpdate(() => {
      expect(cb).toHaveBeenCalled()
    }).then(done)
  })

  it('triggers event inactive when other Modals becomes active', done => {
    vm = createVue({
      template: `
        <div>
          <vk-modal ref="modal1" :show="true" :transition="false">Modal 1</vk-modal>
          <vk-modal ref="modal2" :show="show" :transition="false">Modal 2</vk-modal>
        </div>
      `,
      data: () => ({
        show: false
      })
    }, true)
    const cb = jest.fn()
    vm.$refs.modal1.$on('inactive', cb)
    vm.show = true
    waitForUpdate(() => {
      expect(cb).toHaveBeenCalled()
    }).then(done)
  })
})
