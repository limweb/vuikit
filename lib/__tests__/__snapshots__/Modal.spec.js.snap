exports[`Modal renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-modal",
        "style": Array [
          "display:",
          "block;",
        ],
      },
      "child": Array [
        Object {
          "attr": Object {
            "class": Array [
              "uk-modal-dialog",
              "uk-modal-dialog-large",
              "uk-modal-dialog-lightbox",
              "uk-modal-dialog-blank",
            ],
          },
          "child": Array [
            Object {
              "node": "text",
              "text": "Modal",
            },
          ],
          "node": "element",
          "tag": "div",
        },
      ],
      "node": "element",
      "tag": "div",
    },
  ],
  "node": "root",
}
`;
